examples/chip-tool/commands/common/Command.cpp

```diff
@@ -32,6 +32,10 @@
 #include <lib/support/ScopedBuffer.h>
 #include <lib/support/logging/CHIPLogging.h>
 
+#define NO_OPTIONAL_ARGS 0
+#define OPTIONAL_ARGS_OFFSET 2
+#define OPTIONAL_VALUE_OFFSET 1
```

Comment:
This is because the value is right after the name?

I think it might be clearer to just have functions for getting the Nth optional arg name and the Nth optional arg value (which take argv, the number of non-optional args expected, and N).